File: Labs/Lab4/Lab4.py
class TreeNode:

  def __init__(self, val=0, left=None, right=None):
    self.val = val
    self.left = left
    self.right = right


def invert(currNode):
  if currNode is not None and currNode.left is not None and currNode.right is not None:
    tempNode = currNode.right
    currNode.right = currNode.left
    currNode.left = tempNode
    invert(currNode.left)
    invert(currNode.right)


def main():
  node3 = TreeNode(3)
  node4 = TreeNode(4)
  node5 = TreeNode(5)
  node6 = TreeNode(6)
  node2 = TreeNode(2, node3, node4)
  node8 = TreeNode(8, node5, node6)
  root = TreeNode(1, node2, node8)

  if (root != None and root.left != None and root.right != None):
    print(root.val)
    print(root.left.val)
    print(root.left.left.val)
    print(root.left.right.val)
    print(root.right.val)
    print(root.right.right.val)
    print(root.right.left.val)
    print()

    invert(root)

    print(root.val)
    print(root.left.val)
    print(root.left.left.val)
    print(root.left.right.val)
    print(root.right.val)
    print(root.right.right.val)
    print(root.right.left.val)


if __name__ == "__main__":
  main()

File: Labs/Lab4/Lab4.java

public class Lab4 {
	public static void invert(TreeNode currNode){
		if(currNode != null && currNode.left != null && currNode.right != null){
			TreeNode tempNode = currNode.right;
			currNode.right = currNode.left;
			currNode.left = tempNode;
			invert(currNode.left);
			invert(currNode.right);
		}
	}
	
		public static void main(String[] args) {
			TreeNode node3 = new TreeNode(3);
			TreeNode node4 = new TreeNode(4);
			TreeNode node5 = new TreeNode(5);
			TreeNode node6 = new TreeNode(6);
			TreeNode node2 = new TreeNode(2, node3, node4);
			TreeNode node8 = new TreeNode(8, node5, node6);
			TreeNode root = new TreeNode(1, node2, node8);
			System.out.println(root.val);
			System.out.println(root.left.val);
			System.out.println(root.left.left.val);
			System.out.println(root.left.right.val);
			System.out.println(root.right.val);
			System.out.println(root.right.right.val);
			System.out.println(root.right.left.val);
			System.out.println();
			invert(root);
			System.out.println(root.val);
			System.out.println(root.left.val);
			System.out.println(root.left.left.val);
			System.out.println(root.left.right.val);
			System.out.println(root.right.val);
			System.out.println(root.right.right.val);
			System.out.println(root.right.left.val);
		}
}


